Description: Enable compatibility with Linux 6.4+
 class_create no longer takes a 'struct module *' as a first argument, and
 instead expects just a string. When building on 6.4+, use the new
 class_create syntax.
Author: Aaron Rainbolt <arraybolt3@ubuntu.com>
Origin: upstream, https://gitlab.com/tuxedocomputers/development/packages/tuxedo-drivers/-/commit/04112f65ec4099d43bca8c00acc61f3a2c0e185f
Bug-Kfocus: https://github.com/kfocus/kfocus-source/issues/17
Last-Update: 2024-01-11
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/tuxedo_io/tuxedo_io.c
+++ b/src/tuxedo_io/tuxedo_io.c
@@ -827,7 +827,13 @@ static int __init tuxedo_io_init(void)
 		pr_err("Failed to add cdev\n");
 		unregister_chrdev_region(tuxedo_io_device_handle, 1);
 	}
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
 	tuxedo_io_device_class = class_create(THIS_MODULE, "tuxedo_io");
+#else
+	tuxedo_io_device_class = class_create("tuxedo_io");
+#endif
+
 	device_create(tuxedo_io_device_class, NULL, tuxedo_io_device_handle, NULL, "tuxedo_io");
 	pr_debug("Module init successful\n");
 	
